packages/meditrak-app-server/src/app/createApp.ts

```diff
@@ -3,13 +3,35 @@
  * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
  */
 import { TupaiaDatabase } from '@tupaia/database';
-import { MicroServiceApiBuilder } from '@tupaia/server-boilerplate';
+import { handleWith, MicroServiceApiBuilder } from '@tupaia/server-boilerplate';
+import {
+  ForwardingAuthHandler,
+  getBaseUrlsForHost,
+  LOCALHOST_BASE_URLS,
+  TupaiaApiClient,
+} from '@tupaia/api-client';
+import { AuthRequest, AuthRoute } from '../routes';
 
 /**
  * Set up express server with middleware,
  */
 export function createApp() {
-  const app = new MicroServiceApiBuilder(new TupaiaDatabase()).build();
+  const app = new MicroServiceApiBuilder(new TupaiaDatabase())
+    .middleware((req, res, next) => {
+      const baseUrls =
+        process.env.NODE_ENV === 'test' ? LOCALHOST_BASE_URLS : getBaseUrlsForHost(req.hostname);
+      const context = {
+        services: new TupaiaApiClient(
+          new ForwardingAuthHandler(req.headers.authorization),
```

Comment:
Just using a forwarding auth handler for now, as the `TupaiaApi` in the `meditrak-app` manages what header to send through currently. However, in the future it might be best to make the `meditrak-app-server` responsible for passing through the appropriate header (be it the user's bearer token, or it's own api-client auth header)